packages/cli/src/bundle/metro.ts

```diff
@@ -10,40 +10,20 @@ import type { TSProjectInfo } from "../types";
 import type { BundleConfig } from "./types";
 
 /**
- * Run the Metro bundler.
+ * Create Metro bundler arguments from a bundle configuration.
  *
- * @param tsservice TypeScript service to use for type-checking (when enabled)
- * @param metroConfig Metro configuration
  * @param bundleConfig Bundle configuration
+ * @returns Metro bundle arguments
  */
-export async function metroBundle(
-  tsservice: Service,
-  metroConfig: ConfigT,
+export function createMetroBundleArgs(
   bundleConfig: BundleConfig
-): Promise<void> {
-  const targetPlatform = bundleConfig.platform;
-
-  info(`Bundling ${targetPlatform}...`);
-
-  const {
-    entryPath,
-    distPath,
-    assetsPath,
-    bundlePrefix,
-    bundleEncoding,
-    sourceMapSourceRootPath,
-    detectCyclicDependencies,
-    detectDuplicateDependencies,
-    typescriptValidation,
-    experimental_treeShake,
-  } = bundleConfig;
+): MetroBundleArgs {
+  const { platform, distPath } = bundleConfig;
```

Comment:
We should extract everything we need here once, then reference them further down:
```suggestion
  const {
    assetsPath: assetsDest,
    bundleEncoding,
    bundlePrefix,
    entryPath: entryFile,
    dev,
    distPath,
    platform,
    sourceMapSourceRootPath: sourcemapSourcesRoot,
  } = bundleConfig;
```

Reply:
I had that but then went back to only extracting what I used more than once. My thinking is that I don't need storage space for the single-use params and I only want to read them once. I realize I am making a lot of assumptions about how the JSVM works, so if I've gone wrong, let me know... And then there's the calculus of source-code size and how much will be minified. I don't have enough exp with that to know how it will impact this, either.

How do you see one choice vs. the other when considering storage space and number of reads? Are they real concerns?

Reply:
I should probably have prefixed this with "Nit" 😛 

I think it's mostly subjective at this point. I like having everything destructured once so we don't make separate lookups. Any savings (if any) would be insignificant. I had a look at the output, and it seems like the statement is preserved, so I am _assuming_ that the JSVM can make some optimizations under the hood if we used destructuring. Minified, I think this syntax will be smaller since you don't have to keep referencing `bundleConfig`.